cmd/ebrelayer/relayer/cosmosReplay.go

```diff
@@ -0,0 +1,201 @@
+package relayer
+
+import (
+	"context"
+	"log"
+	"math/big"
+
+	cosmosbridge "github.com/Sifchain/sifnode/cmd/ebrelayer/contract/generated/bindings/cosmosbridge"
+	"github.com/Sifchain/sifnode/cmd/ebrelayer/txs"
+	"github.com/Sifchain/sifnode/cmd/ebrelayer/types"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	tmClient "github.com/tendermint/tendermint/rpc/client/http"
+	"go.uber.org/zap"
+)
+
+// ProphecyLifeTime signature info life time on chain
+const ProphecyLifeTime = 520000
+
+// ReplayCosmosBurnLock the missed burn lock events from cosmos
+func (sub CosmosSub) ReplayCosmosBurnLock(txFactory tx.Factory) {
+	client, err := tmClient.New(sub.TmProvider, "/websocket")
+	if err != nil {
+		log.Printf("failed to initialize a client, error as %s\n", err)
+		return
+	}
+
+	if err := client.Start(); err != nil {
+		log.Printf("failed to start a client, error as %s\n", err)
+		return
+	}
+
+	fromBlock, toBlock, err := GetScannedBlockScope(client)
+	if err != nil {
+		log.Printf("failed to get the scaned block scope, error as %s\n", err)
+		return
+	}
+
+	accAddr, err := GetAccAddressFromKeyring(txFactory.Keybase(), sub.ValidatorName)
+	if err != nil {
+		log.Printf("failed to get the account address, error as %s\n", err)
+		return
+	}
+
+	ProphecyClaims := sub.getAllSignSigature(client, accAddr, fromBlock, toBlock)
+
+	sub.ReplayCosmosBurnLockWithBlocks(txFactory, client, accAddr, ProphecyClaims, fromBlock, toBlock)
+}
+
+// ReplayCosmosBurnLockWithBlocks replay the missed burn lock events from cosmos
```

Comment:
This comment needs to actually describe what is going on in this function as the function itself is confusing.